webapp/channels/src/components/post_view/commented_on/commented_on.tsx

```diff
@@ -5,46 +5,47 @@ import React, {memo} from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import {isMessageAttachmentArray} from '@mattermost/types/message_attachments';
-import type {Post} from '@mattermost/types/posts';
-import type {UserProfile as UserProfileType} from '@mattermost/types/users';
 
+import {usePost} from 'components/common/hooks/usePost';
+import {useUser} from 'components/common/hooks/useUser';
 import CommentedOnFilesMessage from 'components/post_view/commented_on_files_message';
 import UserProfile from 'components/user_profile';
 
 import {stripMarkdown} from 'utils/markdown';
 import * as Utils from 'utils/utils';
 
 type Props = {
-    enablePostUsernameOverride?: boolean;
-    parentPostUser?: UserProfileType;
     onCommentClick?: React.EventHandler<React.MouseEvent>;
-    post: Post;
+    rootId: string;
 };
 
-function CommentedOn({post, parentPostUser, onCommentClick}: Props) {
-    const makeCommentedOnMessage = () => {
-        let message: React.ReactNode = '';
-        if (post.message) {
-            message = Utils.replaceHtmlEntities(post.message);
-        } else if (post.file_ids && post.file_ids.length > 0) {
-            message = (
-                <CommentedOnFilesMessage parentPostId={post.id}/>
-            );
-        } else if (isMessageAttachmentArray(post.props?.attachments) && post.props.attachments.length > 0) {
-            const attachment = post.props.attachments[0];
-            const webhookMessage = attachment.pretext || attachment.title || attachment.text || attachment.fallback || '';
-            message = Utils.replaceHtmlEntities(webhookMessage);
-        }
+function CommentedOn({onCommentClick, rootId}: Props) {
+    const rootPost = usePost(rootId);
+    const rootPostUser = useUser(rootPost?.user_id ?? '');
 
-        return message;
-    };
-
-    const message = makeCommentedOnMessage();
-    const parentPostUserId = parentPostUser?.id ?? '';
+    let message: React.ReactNode = '';
+    if (!rootPost) {
```

Comment:
Before the user/post are loaded, this will now render "Commented on someone's message: Loading..." as a placeholder instead of the `CommentedOn` not being rendered at all. That also means that, if the app fails to load the data, *something* will still be rendered instead of confusing the user by making them think the post is part of the wrong thread